types/capella/state.yaml

```diff
@@ -0,0 +1,109 @@
+Capella:
+  BeaconState:
+    type: object
+    description: "The [`BeaconState`](https://github.com/ethereum/consensus-specs/blob/master/specs/capella/beacon-chain.md#beaconstate) object from the Eth2.0 Capella spec."
+    properties:
+      genesis_time:
+        $ref: "../primitive.yaml#/Uint64"
+      genesis_validators_root:
+        $ref: "../primitive.yaml#/Root"
+      slot:
+        $ref: "../primitive.yaml#/Uint64"
+      fork:
+        $ref: "../misc.yaml#/Fork"
+      latest_block_header:
+        $ref: "../block.yaml#/BeaconBlockHeader"
+      block_roots:
+        type: array
+        items:
+          allOf:
+            - $ref: '../primitive.yaml#/Root'
+        minItems: 8192
+        maxItems: 8192
+      state_roots:
+        type: array
+        items:
+          allOf:
+            - $ref: '../primitive.yaml#/Root'
+        minItems: 8192
+        maxItems: 8192
+      historical_roots:
+        type: array
+        items:
+          allOf:
+            - $ref: '../primitive.yaml#/Root'
+        maxItems: 16777216
+      eth1_data:
+        $ref: "../eth1.yaml#/Eth1Data"
+      eth1_data_votes:
+        type: array
+        items:
+          allOf:
+            - $ref: '../eth1.yaml#/Eth1Data'
+        maxItems: 1024
+      eth1_deposit_index:
+        $ref: "../primitive.yaml#/Uint64"
+      validators:
+        type: array
+        maxItems: 1099511627776
+        items:
+          allOf:
+            - $ref: '../validator.yaml#/Validator'
+      balances:
+        type: array
+        description: "Validator balances in gwei"
+        maxItems: 1099511627776
+        items:
+          allOf:
+            - $ref: '../primitive.yaml#/Uint64'
```

Comment:
found a concrete example of not using allOf:
```
      items:
        $ref: '../primitive.yaml#/Uint64'
```

Reply:
I haven't put comments everywhere, but at least this is the idea...

Reply:
Thanks, updated,